### ISSUE TITLE
How to use brush for zooming into the same graph?

### ISSUE THREAD

**Issue by viky293** [esnet/react-timeseries-charts#87]
As per the example, the brush component can be used with a separate graph which modifies the time range for associated chart. However traditionally people would expect to drag and select the time range on the same graph.

Something like here https://www.amcharts.com/demos/line-chart-with-scroll-and-zoom/?theme=chalk

I could implement it with a small modification to handleMouseUp event handler function, as I only want to update the chart only when selection is done.

```
handleMouseUp: function handleMouseUp(e) {
        e.preventDefault();
        document.removeEventListener("mouseover", this.handleMouseMove);
        document.removeEventListener("mouseup", this.handleMouseUp);

        this.setState({
            isBrushing: false,
            brushingInitializationSite: null,
            initialBrushBeginTime: null,
            initialBrushEndTime: null,
            initialBrushXYPosition: null
        });
        //viky293 clear the selection and call the callback
        if (this.props.onSeclectComplete) {
            this.props.onSeclectComplete();
             this.props.onTimeRangeChanged(null);
        }
    },
```

Another way is we can provide a callback on selectComplete so as to perform any updates only when select is complete.

Is there a way I can extend the component to override these event handlers?
If this seems to be a valid use case I can fork and create a pull request.

**Comment by pjm17971** [comment]
Thanks for the thought on this!

Pretty much the founding goal of the library was to fully support pan and zoom, which basically means that the action of dragging across a chart is semi-reserved for that. The goal was closest to this: https://www.amcharts.com/demos/date-based-data/

Of course you can turn that on and off, so we could support both behaviors. And I agree, the drag and release zoom is pretty common. I also agree this is a lot simpler now since the brushing is native react code (it was a d3 brush until recently). Also, I know of at least one other group that would appreciate this.

So basically, if you want to go ahead and try this and send a PR, please do. 

Thanks!

**Comment by viky293** [comment]
If I make changes to the JSX source file under /src/components shouldn't the build process auto compile and produce equivalent .JS file under /lib/components ? 

I made the corresponding changes in JSX source file but it doesn't reflect in the output's, to be able to send PR is that necessary?

**Comment by pjm17971** [comment]
`npm run build` should rebuild it for npm distribution.

**Comment by viky293** [comment]
Not sure why am I not seeing the changes reflected in final build inside lib/components/ I have already tired `npm run build `, created a pull request with the changes in /src/components/brush.js 
Happy to do the needful.

**Comment by pjm17971** [comment]
I'll rebuild it before I push a release, which won't be for a little while. But first, can you address the comments? Thanks for taking the time to think about this and suggest the addition.

**Comment by viky293** [comment]
Would this be included in the next release? I do not see it in the current version.

**Comment by pjm17971** [comment]
Are you still not seeing it? It looks like from the commit history that it was merged in before the 0.11 release. Let me know if there's still a problem and I'll look into it further.
